### ISSUE TITLE
Count number of leaves in the ensemble gradient BoostedTree

### ISSUE THREAD

**Issue by samanemami** [tensorflow/tensorflow#53871]
This is an issue related to the performance of [TensorFlow.BoostedTree](https://www.tensorflow.org/api_docs/python/tf/estimator/BoostedTreesClassifier)

**System information**
- TensorFlow version (you are using): 2.7.0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
Is there any method that returns the number of entire leaves in the built ensemble?
If the feature already exists, would you please introduce it a bit more here, if not could we develop it?

**Will this change the current api? How?**
No

**Who will benefit with this feature?**
Users of the tensorflow

**Comment by sushreebarsa** [comment]
@samanemami Could you please refer to the [thread](https://github.com/scikit-learn/scikit-learn/discussions/22270#discussioncomment-2024510),[ BoostedTreesClassifier](https://www.tensorflow.org/api_docs/python/tf/estimator/BoostedTreesClassifier) and let us know if it helps?

**Comment by samanemami** [comment]
The [thread](https://github.com/scikit-learn/scikit-learn/discussions/22270#discussioncomment-2024510) is related to Sklearn, and all methods are different in TensorFlow.

Regarding the [BoostedTreesClassifier](https://www.tensorflow.org/api_docs/python/tf/estimator/BoostedTreesClassifier), I could not find any related method!

**Comment by samanemami** [comment]
I already know how to return leaves in `Sklearn`, `LightGBM`, or `XGBoost`.
But with the `tf`, I could not find any method!

**Comment by sachinprasadhs** [comment]
`BoostedTreesClassifier` is depreciated, you can refer to [TensorFlow Decision Forests](https://www.tensorflow.org/decision_forests) and for contribution you can check in the github repo `tensorflow/decision-forests` [here](https://github.com/tensorflow/decision-forests).
